### ISSUE TITLE
Perform worser than ORB3-SLAM on my own collected data.

### ISSUE THREAD

**Issue by Chanchanhan** [MAC-VO/MAC-VO#26]
Thanks for your amazing job!

I collect dataset of my own stereo cameras with eurco format.  I use config file 'MACVO_Performant.yaml' and replace the intrinsic and baseline with code:

```c++
class EuRoC_StereoSequence(SequenceBase[StereoFrame]):
    @classmethod
    def name(cls) -> str: return "EuRoC_NoIMU"

    def __init__(self, config: SimpleNamespace | dict[str, Any]) -> None:
        cfg = self.config_dict2ns(config)
        self.seqRoot  = Path(cfg.root)
        
        # ref: https://github.com/raulmur/ORB_SLAM2/blob/master/Examples/Stereo/EuRoC.yaml
        # in this file only bl * fx is provided , the baseline here is derived by bf/fx
        self.baseline = 0.121 //my camera baseline
        
        # Left Camera
        l_sensor_config, _ = load_config(Path(self.seqRoot, "cam0", "sensor.yaml"))
        
        self.width = l_sensor_config.resolution[0]
        self.height = l_sensor_config.resolution[1]
        
        T_BS_lcam = np.array(l_sensor_config.T_BS.data).reshape(4, 4)
        self.ImageL = EurocMonocularDataset(
            Path(self.seqRoot, "cam0", "data"), 
            K=self.build_intrinsic(l_sensor_config.intrinsics),
            T_BS=T_BS_lcam,
            undistort=np.array(l_sensor_config.distortion_coefficients)
        )
        
        # Right Camera
        r_sensor_config, _ = load_config(Path(self.seqRoot, "cam1", "sensor.yaml"))
        T_BS_rcam = np.array(r_sensor_config.T_BS.data).reshape(4, 4)
        self.ImageR = EurocMonocularDataset(
            Path(self.seqRoot, "cam1", "data"),
            K=self.build_intrinsic(r_sensor_config.intrinsics),
            T_BS=T_BS_rcam,
            undistort=np.array(r_sensor_config.distortion_coefficients)
        )
        
        # Sync left-right camera
        rectified_K = self.sync_LR(self.ImageL, self.ImageR)
        self.K = torch.tensor(rectified_K[:3, :3], dtype=torch.float).unsqueeze(0)
        self.cam_timestamps = self.ImageL.cam_timestamps
        assert len(self.ImageL) == len(self.ImageR), f"ImageL={len(self.ImageL)} and ImageR={len(self.ImageR)} is not sync as expected."
        
        # Setup metadata
        self.T_BS_lcam = pp.from_matrix(
            torch.tensor(T_BS_lcam, dtype=torch.float32).unsqueeze(0), pp.SE3_type
        ) @ NED2EDN.unsqueeze(0)
        
        # Load ground truth pose
        if cfg.gt_pose:
            self.gt_pose_data, self.cam_time_mask = load_EurocGTPose(
                Path(self.seqRoot, "state_groundtruth_estimate0/data.csv"),
                self.ImageL.cam_timestamps
            )
            self.ImageL.apply_mask(self.cam_time_mask)
            self.ImageR.apply_mask(self.cam_time_mask)
        else:
            self.gt_pose_data = None
        
        super().__init__(len(self.ImageL))
``` 

However, the result of MAC-VO  is much worse than ORB3-SLAM's while I input same intrinsic and baseline. 

<img width="2011" height="831" alt="Image" src="https://github.com/user-attachments/assets/2950fe00-4512-4bb0-8978-34965c4bd8b4" />

The dataset is about camera moves around table : 

<img width="1278" height="360" alt="Image" src="https://github.com/user-attachments/assets/413becb8-3daf-4ba0-9e5c-b1b16cc05452" />

The ground truth trajectroy is colleted by auroc tag.

<img width="3173" height="1711" alt="Image" src="https://github.com/user-attachments/assets/b314fcaa-3dc2-4521-9d6b-ef6cd9c6029c" />

Can you give me same help about  caused reason ? Or which pramaters should I adjust for better performace of MAV-VO?

**Comment by haleqiu** [comment]
> > [@Chanchanhan](https://github.com/Chanchanhan) Quick question. Why there is a discrepancy between the gt and macvo in terms of the initialization.
> > 
> > Also I found ORB "jump" a lot in your figure.
> 
> The  discrepancy between the gt and macvo in terms of the initialization is caused by full traj align by evo, if I use '--align_origin' , the result of mac-vo will be worse. 
> 
> <img width="3281" height="1737" alt="Image" src="https://github.com/user-attachments/assets/3588ac67-4281-4fd9-b698-dfee71a5d6ed" />
> 
> And  orb3 fail to track local map sometimes,  which make the trajectory jumped.
> 

Hi just a quick response. This is not worse. The coordinate between the gt and macvo is different. The default coordinate of macvo is NED.

For the comparison, if you only calculate the position error of the orb slam key frame. I think it will be smaller. A fair comparison is either 1. the same selected frames. 2. You should use the all key frame api from ORB SLAM.

**Comment by Chanchanhan** [comment]
> > > [@Chanchanhan](https://github.com/Chanchanhan) Quick question. Why there is a discrepancy between the gt and macvo in terms of the initialization.
> > > Also I found ORB "jump" a lot in your figure.
> > 
> > 
> > The  discrepancy between the gt and macvo in terms of the initialization is caused by full traj align by evo, if I use '--align_origin' , the result of mac-vo will be worse.
> > <img alt="Image" width="2000" height="1737" src="https://private-user-images.githubusercontent.com/19565462/466264533-3588ac67-4281-4fd9-b698-dfee71a5d6ed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI1NDcxODgsIm5iZiI6MTc1MjU0Njg4OCwicGF0aCI6Ii8xOTU2NTQ2Mi80NjYyNjQ1MzMtMzU4OGFjNjctNDI4MS00ZmQ5LWI2OTgtZGZlZTcxYTVkNmVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE1VDAyMzQ0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2MjRmMTZmOTE1MjJkZjczZjRmODRlZjIwNjMxNzBjM2NmZDIxMDE5M2VmNWE5ZThiMGMwYzMwYmZmYmQ5NGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hU3o9Oo_578ICCHSLz8jCHOJqlAKrP4ckI57QnBXaEw">
> > And  orb3 fail to track local map sometimes,  which make the trajectory jumped.
> 
> Hi just a quick response. This is not worse. The coordinate between the gt and macvo is different. The default coordinate of macvo is NED.
> 
> For the comparison, if you only calculate the position error of the orb slam key frame. I think it will be smaller. A fair comparison is either 1. the same selected frames. 2. You should use the all key frame api from ORB SLAM.

Hello, I filter the traj of MAC-VO, now the frame of traj by mac-vo is same as orb's.
But the rmse still worse than orb3.

<img width="3217" height="1784" alt="Image" src="https://github.com/user-attachments/assets/b4cc6133-8a78-4914-8fd0-067ba13f3da2" />

<img width="1517" height="417" alt="Image" src="https://github.com/user-attachments/assets/ae56bb5c-d841-4002-9839-fb8676c55277" />

**Comment by Chanchanhan** [comment]
Thanks for fast relpy. I collected more dataset, and I found most case the accuracy of MAC-VO' traj is about 20%-30% higher than orb3, but  still left some case ORB3 perform better.  So maybe MAC-VO is not always better than orb3?

**Comment by haleqiu** [comment]
Just as @MarkChenYutian  pointed out, ORB-SLAM is a complete SLAM system—it handles keyframe selection, bundle adjustment, and loop closure—whereas MAC-VO provides only two-frame odometry. Comparing their ape isn’t entirely fair without accounting for ORB-SLAM’s loop-closure corrections. Also, using AUROC-based ground truth may be misleading here, since neither algorithm actually “fails” in your example.

In the sequence you showed, ORB-SLAM exhibits noticeable jitter and occasional “jumps” (not sure is due to lost tracking), which would be unacceptable for our lab’s robots—these spikes can cause the planner and controller to crash. MAC-VO delivers the expected performance.

If you need a full SLAM pipeline, stay tuned: we’re currently developing MAC-SLAM, which will add keyframe management and loop-closure to MAC-VO.

**Comment by Chanchanhan** [comment]
> MAC-SLAM

Thanks for your reply. Look forward for your MAC-SLAM
